benchtable.js

```diff
@@ -1,218 +1,235 @@
-/**
- * Module dependencies.
- */
-
-var Benchmark = require('benchmark');
-var Table = require('cli-table');
-
-/**
- * Table constructor.
- */
-
-function BenchTable(name, options) {
-  var self = this;
-
-  Benchmark.Suite.call(self, name, options);
-
-  self._functions = [];
-  self._functionNames = [];
-  self._functionOptions = [];
-  self._inputs = [];
-  self._inputNames = [];
-
-  self._results = {};
-
-  // mappings from benchmark names to function and input indices
-  // as { funcIdx, inputIdx }
-  self._mappings = {};
-
-  // if transposed, then functions are in columns and inputs are in rows
-  self._transposed = options && options.isTransposed;
-
-  /**
-   * Use the 'cycle' event to store results and build a table
-   * after all BenchTable functions are evaluated for all inputs.
-   **/
-
-  self.on('cycle', function(event) {
-    var key = event.target.name;
-    var self = this;
-    var worst_idx, best_idx, worst, best, curr;
-    var i, j;
-    var funName, inputName, item;
+"use strict";
+
+// Module dependencies.
+const Benchmark = require('benchmark');
+const Table = require('cli-table');
+
+module.exports = class BenchTable extends Benchmark.Suite {
+    /**
+     * BenchTable
+     * The `BenchTable` constructor. `BenchTable` is extended from [Benchmark.Suite](http://benchmarkjs.com/docs#Suite).
+     *
+     * @name BenchTable
+     * @function
+     * @param {String} name  A name to identify the suite.
+     * @param {String} options Options object. A special boolean option `isTransposed`
+     * is available for Benchtable–if set to `true`, the output ASCII table will be
+     * transposed: the benchmarked functions will be organized into columns, while
+     * inputs will be organized into rows. This is useful for situations where the
+     * number of inputs is large, and the number of functions is small. This option
+     * defaults to `false`.
+     */
+    constructor (name, options) {
+
+        super(name, options);
```

Comment:
> I think I see a new version of benchmark.js published on npm

@izuzak Great! I checked that when I woke up (6 hours ago) and saw that is not published. :smile:

Done in e6f85f86de7e56a1d6f01359dbf0d5da3586a86c and fixed the small issues after migration in the latest commits (replaced `pluck` with `map` in the example files and rebuilt the docs, use [`color-it`](https://github.com/ionicabizau/node-color-it) for colors).

Of course, check it out on your side and give me feedback. :grin: